keepalived/check/check_dns.c

```diff
@@ -388,6 +388,26 @@ dns_dump(void *data)
 	log_message(LOG_INFO, "   Name = %s", dns_check->name);
 }
 
+static int
+dns_compare(void *a, void *b)
+{
+	dns_check_t *old = CHECKER_DATA(a);
+	dns_check_t *new = CHECKER_DATA(b);
+
+	if (compare_conn_opts(CHECKER_CO(a), CHECKER_CO(b)) != 0)
+		goto err;
+	if (old->retry != new->retry)
+		goto err;
```

Comment:
I'm not convinced that different values of retry make this a different checker. Certainly if the old checker had failed, it seems likely that a new one differing only in this value would also fail.